--- nss-3.12.2.99.3/mozilla/security/nss/lib/freebl/Makefile-save	2009-04-02 08:46:32.083530732 -0700
+++ nss-3.12.2.99.3/mozilla/security/nss/lib/freebl/Makefile	2009-04-02 08:46:51.740542226 -0700
@@ -133,7 +133,7 @@
 ifeq ($(OS_TARGET),Linux)
 ifeq ($(CPU_ARCH),x86_64)
     ASFILES  = arcfour-amd64-gas.s mpi_amd64_gas.s
-    ASFLAGS += -march=opteron -m64 -fPIC
+    ASFLAGS += -march=opteron -m64 -fPIC -Wa,--noexecstack
     DEFINES += -DNSS_BEVAND_ARCFOUR -DMPI_AMD64 -DMP_ASSEMBLY_MULTIPLY
     DEFINES += -DNSS_USE_COMBA
     DEFINES += -DMP_CHAR_STORE_SLOW -DMP_IS_LITTLE_ENDIAN

