diff -Nur nss-3.4.rc1.orig/mozilla/security/coreconf/Linux.mk nss-3.4.rc1/mozilla/security/coreconf/Linux.mk
--- nss-3.4.rc1.orig/mozilla/security/coreconf/Linux.mk	Fri Feb 15 23:53:12 2002
+++ nss-3.4.rc1/mozilla/security/coreconf/Linux.mk	Thu Apr 11 20:48:58 2002
@@ -121,8 +121,8 @@
 DSO_LDOPTS		= -shared
 DSO_LDFLAGS		=
 
-# INCLUDES += -I/usr/include -Y/usr/include/linux
 G++INCLUDES		= -I/usr/include/g++
+INCLUDES += -I/usr/include/nspr
 
 #
 # Always set CPU_TAG on Linux, OpenVMS, WINCE.
diff -Nur nss-3.4.rc1.orig/mozilla/security/coreconf/ruleset.mk nss-3.4.rc1/mozilla/security/coreconf/ruleset.mk
--- nss-3.4.rc1.orig/mozilla/security/coreconf/ruleset.mk	Thu Feb 21 22:23:14 2002
+++ nss-3.4.rc1/mozilla/security/coreconf/ruleset.mk	Thu Apr 11 20:48:26 2002
@@ -114,7 +114,7 @@
 
 ifdef LIBRARY_NAME
     ifndef LIBRARY
-	LIBRARY        = $(OBJDIR)/$(LIB_PREFIX)$(LIBRARY_NAME).$(LIB_SUFFIX)
+	LIBRARY        = $(OBJDIR)/$(LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(LIB_SUFFIX)
     endif
     ifndef SHARED_LIBRARY
 	SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(JDK_DEBUG_SUFFIX).$(DLL_SUFFIX)
diff -Nur nss-3.4.rc1.orig/mozilla/security/nss/cmd/platlibs.mk nss-3.4.rc1/mozilla/security/nss/cmd/platlibs.mk
--- nss-3.4.rc1.orig/mozilla/security/nss/cmd/platlibs.mk	Sat Feb 16 00:38:41 2002
+++ nss-3.4.rc1/mozilla/security/nss/cmd/platlibs.mk	Thu Apr 11 20:45:49 2002
@@ -82,21 +82,21 @@
 else
 
 # $(PROGRAM) has explicit dependencies on $(EXTRA_LIBS)
-CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)freebl.$(LIB_SUFFIX)
+CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)freebl_3.$(LIB_SUFFIX)
 ifdef MOZILLA_SECURITY_BUILD
 	CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)crypto.$(LIB_SUFFIX)
 endif
 ifdef MOZILLA_BSAFE_BUILD
 	CRYPTOLIB+=$(DIST)/lib/$(LIB_PREFIX)bsafe.$(LIB_SUFFIX)
-	CRYPTOLIB+=$(DIST)/lib/$(LIB_PREFIX)freebl.$(LIB_SUFFIX)
+	CRYPTOLIB+=$(DIST)/lib/$(LIB_PREFIX)freebl_3.$(LIB_SUFFIX)
 endif
 EXTRA_LIBS += \
-	$(DIST)/lib/$(LIB_PREFIX)smime.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)ssl.$(LIB_SUFFIX) \
+	$(DIST)/lib/$(LIB_PREFIX)smime3.$(LIB_SUFFIX) \
+	$(DIST)/lib/$(LIB_PREFIX)ssl3.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)jar.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)zlib.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)nss.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)ssl.$(LIB_SUFFIX) \
+	$(DIST)/lib/$(LIB_PREFIX)nss3.$(LIB_SUFFIX) \
+	$(DIST)/lib/$(LIB_PREFIX)ssl3.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)sectool.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)pkcs12.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)pkcs7.$(LIB_SUFFIX) \
@@ -104,11 +104,11 @@
 	$(DIST)/lib/$(LIB_PREFIX)pk11wrap.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)cryptohi.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)certhi.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)nsspki.$(LIB_SUFFIX) \
+	$(DIST)/lib/$(LIB_PREFIX)nsspki3.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)pk11wrap.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)softokn.$(LIB_SUFFIX) \
+	$(DIST)/lib/$(LIB_PREFIX)softokn3.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)certdb.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)nsspki.$(LIB_SUFFIX) \
+	$(DIST)/lib/$(LIB_PREFIX)nsspki3.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)nssdev.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)nssb.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)swfci.$(LIB_SUFFIX) \
diff -Nur nss-3.4.rc1.orig/mozilla/security/nss/lib/fortcrypt/swfort/pkcs11/Makefile nss-3.4.rc1/mozilla/security/nss/lib/fortcrypt/swfort/pkcs11/Makefile
--- nss-3.4.rc1.orig/mozilla/security/nss/lib/fortcrypt/swfort/pkcs11/Makefile	Fri Feb 15 23:53:46 2002
+++ nss-3.4.rc1/mozilla/security/nss/lib/fortcrypt/swfort/pkcs11/Makefile	Thu Apr 11 20:46:45 2002
@@ -36,7 +36,7 @@
 include $(CORE_DEPTH)/coreconf/config.mk
 include config.mk
 
-CRYPTO_LIB = $(DIST)/lib/$(LIB_PREFIX)freebl.$(LIB_SUFFIX) 
+CRYPTO_LIB = $(DIST)/lib/$(LIB_PREFIX)freebl_3.$(LIB_SUFFIX) 
 #SWCILI    = ../$(OBJDIR)/$(LIB_PREFIX)swfci.$(LIB_SUFFIX)
 
 # can't do this in manifest.mn because OS_TARGET isn't defined there.
@@ -67,7 +67,7 @@
 CRYPTO_LIB = $(DIST)/lib/$(LIB_PREFIX)crypto.$(LIB_SUFFIX)
 endif
 ifdef MOZILLA_BSAFE_BUILD
-CRYPTO_LIB += $(DIST)/lib/$(LIB_PREFIX)bsafe.$(LIB_SUFFIX)
+CRYPTO_LIB += $(DIST)/lib/$(LIB_PREFIX)bsafe_3.$(LIB_SUFFIX)
 CRYPTO_LIB += $(DIST)/lib/$(LIB_PREFIX)freebl.$(LIB_SUFFIX)
 endif
 
@@ -73,11 +73,11 @@
 
 EXTRA_LIBS += \
 	$(DIST)/lib/$(LIB_PREFIX)swfci.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)softokn.$(LIB_SUFFIX) \
+	-L$(DIST)/lib -lsoftokn3 \
 	$(CRYPTO_LIB) \
 	$(DIST)/lib/$(LIB_PREFIX)secutil.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)plc4.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)plds4.$(LIB_SUFFIX) \
+	-lplc4 \
+	-lplds4 \
 	$(NULL)
 
 endif
--- nss-3.4.rc1.orig/mozilla/security/nss/lib/softoken/config.mk	Sun Mar  3 03:34:32 2002
+++ nss-3.4.rc1/mozilla/security/nss/lib/softoken/config.mk	Thu Apr 11 22:13:48 2002
@@ -37,7 +37,7 @@
 endif
 
 # $(PROGRAM) has explicit dependencies on $(EXTRA_LIBS)
-CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)freebl.$(LIB_SUFFIX)
+CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)freebl_3.$(LIB_SUFFIX)
 CRYPTODIR=../freebl
 ifdef MOZILLA_SECURITY_BUILD
 	CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)crypto.$(LIB_SUFFIX)
--- nss-3.4.rc1.orig/mozilla/security/nss/lib/ckfw/builtins/Makefile	Fri Feb 15 23:53:42 2002
+++ nss-3.4.rc1/mozilla/security/nss/lib/ckfw/builtins/Makefile	Thu Apr 11 22:17:58 2002
@@ -54,8 +54,8 @@
 else
 
 EXTRA_LIBS += \
-	$(DIST)/lib/$(LIB_PREFIX)plc4.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)plds4.$(LIB_SUFFIX) \
+	-lplc4 \
+	-lplds4 \
 	$(NULL)
 
 endif
