--- nss-3.10/mozilla/security/nss/cmd/manifest.mn.orig	2005-04-06 23:35:41.000000000 +0200
+++ nss-3.10/mozilla/security/nss/cmd/manifest.mn	2005-04-29 07:38:34.665164688 +0200
@@ -41,7 +41,6 @@
 REQUIRES = nss nspr libdbm
 
 DIRS = lib  \
- zlib \
  addbuiltin \
  atob  \
  bltest \
--- nss-3.8/mozilla/security/nss/cmd/platlibs.mk.orig	Mon May 19 10:09:48 2003
+++ nss-3.8/mozilla/security/nss/cmd/platlibs.mk	Mon May 19 10:16:54 2003
@@ -141,7 +141,7 @@
 endif
 
 JAR_LIBS = $(DIST)/lib/$(LIB_PREFIX)jar.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)zlib.$(LIB_SUFFIX) \
+	-lz \
 	$(NULL)
 
 else # USE_STATIC_LIBS
@@ -219,7 +219,7 @@
 	$(NULL)
 
 JAR_LIBS = $(DIST)/lib/$(LIB_PREFIX)jar.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)zlib.$(LIB_SUFFIX) \
+	-lz \
 	$(NULL)
 endif
 
