--- nss-3.8/mozilla/security/nss/cmd/manifest.mn.orig	Tue Jan 28 18:53:19 2003
+++ nss-3.8/mozilla/security/nss/cmd/manifest.mn	Mon May 19 10:58:27 2003
@@ -37,7 +37,6 @@
 REQUIRES = nss nspr libdbm
 
 DIRS = lib  \
- zlib \
  atob  \
  bltest \
  btoa  \
--- nss-3.8/mozilla/security/nss/cmd/platlibs.mk.orig	Mon May 19 10:09:48 2003
+++ nss-3.8/mozilla/security/nss/cmd/platlibs.mk	Mon May 19 10:16:54 2003
@@ -141,7 +141,7 @@
 endif
 
 JAR_LIBS = $(DIST)/lib/$(LIB_PREFIX)jar.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)zlib.$(LIB_SUFFIX) \
+	-lz \
 	$(NULL)
 
 else # USE_STATIC_LIBS
@@ -219,7 +219,7 @@
 	$(NULL)
 
 JAR_LIBS = $(DIST)/lib/$(LIB_PREFIX)jar.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)zlib.$(LIB_SUFFIX) \
+	-lz \
 	$(NULL)
 endif
 
