diff -Nur nss-3.11.orig/mozilla/security/coreconf/Linux.mk nss-3.11/mozilla/security/coreconf/Linux.mk
--- nss-3.11.orig/mozilla/security/coreconf/Linux.mk	2005-09-16 18:09:23.000000000 +0100
+++ nss-3.11/mozilla/security/coreconf/Linux.mk	2006-01-26 15:14:24.000000000 +0000
@@ -165,12 +165,12 @@
 ARCH			= linux
 
 DSO_CFLAGS		= -fPIC
-DSO_LDOPTS		= -shared $(ARCHFLAG) -z defs
+DSO_LDOPTS		= -shared $(ARCHFLAG) -Wl,-z,defs
 DSO_LDFLAGS		=
 LDFLAGS			+= $(ARCHFLAG)
 
-# INCLUDES += -I/usr/include -Y/usr/include/linux
 G++INCLUDES		= -I/usr/include/g++
+INCLUDES += -I/usr/include/nspr
 
 #
 # Always set CPU_TAG on Linux, OpenVMS, WINCE.
diff -Nur nss-3.11.orig/mozilla/security/coreconf/ruleset.mk nss-3.11/mozilla/security/coreconf/ruleset.mk
--- nss-3.11.orig/mozilla/security/coreconf/ruleset.mk	2005-09-16 18:09:23.000000000 +0100
+++ nss-3.11/mozilla/security/coreconf/ruleset.mk	2006-01-26 15:13:55.000000000 +0000
@@ -118,7 +118,7 @@
 
 ifdef LIBRARY_NAME
     ifndef LIBRARY
-	LIBRARY        = $(OBJDIR)/$(LIB_PREFIX)$(LIBRARY_NAME).$(LIB_SUFFIX)
+	LIBRARY        = $(OBJDIR)/$(LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(LIB_SUFFIX)
     endif
     ifndef SHARED_LIBRARY
 	SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(JDK_DEBUG_SUFFIX).$(DLL_SUFFIX)
diff -Nur nss-3.11.orig/mozilla/security/nss/cmd/platlibs.mk nss-3.11/mozilla/security/nss/cmd/platlibs.mk
--- nss-3.11.orig/mozilla/security/nss/cmd/platlibs.mk	2005-11-15 18:19:18.000000000 +0000
+++ nss-3.11/mozilla/security/nss/cmd/platlibs.mk	2006-01-26 15:13:55.000000000 +0000
@@ -84,19 +84,19 @@
 else
 
 # $(PROGRAM) has explicit dependencies on $(EXTRA_LIBS)
-CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)freebl.$(LIB_SUFFIX)
+CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)freebl3.$(LIB_SUFFIX)
 ifdef MOZILLA_SECURITY_BUILD
 	CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)crypto.$(LIB_SUFFIX)
 endif
 ifdef MOZILLA_BSAFE_BUILD
 	CRYPTOLIB+=$(DIST)/lib/$(LIB_PREFIX)bsafe.$(LIB_SUFFIX)
-	CRYPTOLIB+=$(DIST)/lib/$(LIB_PREFIX)freebl.$(LIB_SUFFIX)
+	CRYPTOLIB+=$(DIST)/lib/$(LIB_PREFIX)freebl3.$(LIB_SUFFIX)
 endif
 EXTRA_LIBS += \
-	$(DIST)/lib/$(LIB_PREFIX)smime.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)ssl.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)nss.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)ssl.$(LIB_SUFFIX) \
+	$(DIST)/lib/$(LIB_PREFIX)smime3.$(LIB_SUFFIX) \
+	$(DIST)/lib/$(LIB_PREFIX)ssl3.$(LIB_SUFFIX) \
+	$(DIST)/lib/$(LIB_PREFIX)nss3.$(LIB_SUFFIX) \
+	$(DIST)/lib/$(LIB_PREFIX)ssl3.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)sectool.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)pkcs12.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)pkcs7.$(LIB_SUFFIX) \
@@ -104,11 +104,11 @@
 	$(DIST)/lib/$(LIB_PREFIX)pk11wrap.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)cryptohi.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)certhi.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)nsspki.$(LIB_SUFFIX) \
+	$(DIST)/lib/$(LIB_PREFIX)nsspki3.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)pk11wrap.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)softokn.$(LIB_SUFFIX) \
+	$(DIST)/lib/$(LIB_PREFIX)softokn3.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)certdb.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)nsspki.$(LIB_SUFFIX) \
+	$(DIST)/lib/$(LIB_PREFIX)nsspki3.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)nssdev.$(LIB_SUFFIX) \
 	$(DIST)/lib/$(LIB_PREFIX)nssb.$(LIB_SUFFIX) \
 	$(CRYPTOLIB) \
diff -Nur nss-3.11.orig/mozilla/security/nss/lib/softoken/config.mk nss-3.11/mozilla/security/nss/lib/softoken/config.mk
--- nss-3.11.orig/mozilla/security/nss/lib/softoken/config.mk	2005-10-19 02:03:31.000000000 +0100
+++ nss-3.11/mozilla/security/nss/lib/softoken/config.mk	2006-01-26 15:13:55.000000000 +0000
@@ -36,7 +36,7 @@
 # ***** END LICENSE BLOCK *****
 
 # $(PROGRAM) has explicit dependencies on $(EXTRA_LIBS)
-CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)freebl.$(LIB_SUFFIX)
+CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)freebl3.$(LIB_SUFFIX)
 CRYPTODIR=../freebl
 ifdef MOZILLA_SECURITY_BUILD
 	CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)crypto.$(LIB_SUFFIX)
diff -Nur nss-3.11.orig/mozilla/security/nss/lib/ssl/config.mk nss-3.11/mozilla/security/nss/lib/ssl/config.mk
--- nss-3.11.orig/mozilla/security/nss/lib/ssl/config.mk	2005-10-19 02:04:16.000000000 +0100
+++ nss-3.11/mozilla/security/nss/lib/ssl/config.mk	2006-01-26 15:13:55.000000000 +0000
@@ -67,7 +67,7 @@
 endif # NS_USE_GCC
 
 # $(PROGRAM) has explicit dependencies on $(EXTRA_LIBS)
-CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)freebl.$(LIB_SUFFIX)
+CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)freebl3.$(LIB_SUFFIX)
 CRYPTODIR=../freebl
 ifdef MOZILLA_SECURITY_BUILD
 	CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)crypto.$(LIB_SUFFIX)
@@ -81,7 +81,7 @@
 else
 
 # $(PROGRAM) has explicit dependencies on $(EXTRA_LIBS)
-CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)freebl.$(LIB_SUFFIX)
+CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)freebl3.$(LIB_SUFFIX)
 CRYPTODIR=../freebl
 ifdef MOZILLA_SECURITY_BUILD
 	CRYPTOLIB=$(DIST)/lib/$(LIB_PREFIX)crypto.$(LIB_SUFFIX)
